diff --git a/funcy/colls.py b/funcy/colls.py
index 4c466fb..eee5d31 100644
--- a/funcy/colls.py
+++ b/funcy/colls.py
@@ -33,7 +33,7 @@ def _factory(coll, mapper=None):
     coll_type = type(coll)
     # Hack for defaultdicts overridden constructor
     if isinstance(coll, defaultdict):
-        item_factory = compose(mapper, coll.default_factory) if mapper and coll.default_factory \
+        item_factory = compose(mapper, coll.default_factory) if mapper \
                        else coll.default_factory
         return partial(defaultdict, item_factory)
     elif isinstance(coll, Iterator):
